File: proytaller2/src/service/EmpleadoService.js
import axios from "axios"

const API_URL = "http://localhost:9090/api/empleado"

const apiEmpleado = axios.create({
    baseURL: API_URL,
    headers: {
        "Content-Type": "application/json"
    }
})

export default {
    
    //añadir empleado
    addEmpleado(empleado){
        return apiEmpleado.post("/agregar", empleado)
    },

    //eliminar empleado
    deleteEmpleado(idEmpleado){
        return apiEmpleado.delete(`/eliminar/${idEmpleado}`)
    },

    //actualizar empleado
    updateEmpleado(idEmpleado, empleado){
        return apiEmpleado.put(`/update/${idEmpleado}`, empleado)
    },

    //buscar empleado por nombre
    buscarEmpleado(nombre){
        return apiEmpleado.get(`/buscar?nombre=${nombre}`)
    },

    //Lista de todos los empleados
    listarEmpleados() {
        return apiEmpleado.get("/ver-empleados");
    },

    //obtener empleado por id
    obtenerEmpleadoPorId(idEmpleado){
        return apiEmpleado.get(`/obtener/${idEmpleado}`)
    }
}

File: proyect/proyect/src/main/java/taller2backend/proyect/service/EmpleadoService.java
package taller2backend.proyect.service;

import java.util.List;
import taller2backend.proyect.entity.Empleado;

public interface EmpleadoService {

    //lisatr todos los empleados
   
    public List<Empleado> obtenerDatos();

    //obtener empleado por id
    public Empleado obtenerEmpeadoPorId(Long idEmnpleado);

    //agregar empleado
    public Empleado agregarEmpleado(Empleado empleado);

    //actualizar empleado
    public Empleado actualizarEmpleado(Long idEmpleado,Empleado empleado);

    //eliminar empleado
    public void eliminarEmpleado(Long idEmpleado);

    //buscar empleado por nombre
    public List<Empleado> buscarEmpleadoPorNombre(String nombre);



}
